### PR TITLE
Development

### PR DESCRIPTION
**Pull request by mikloslorinczi**
Dep implemented
Executor got its own package
Tests are wrapped in Testify's suite
Errors implemented

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// ParseCommand takes the command string and split it with the " " separator.
+// The first element will be the basecommand and the others will be the args.
+func ParseCommand(command string) (string, []string) {
```
Comment (tamaseger81) [inline executor/executor.go]:
```
type Command string

type CommandArguments []string

func ParseCommand(command string) (Command, CommandArguments)
```

File: executor/executor.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	args := commandSlice[1:]
+	return name, args
+}
+
+// GetWriteFile accepts a filename and tries to open it for Writing.
+// It returns a pointer to the file, and an optional error (if it fails to get the required file).
+func GetWriteFile(fileName string) (*os.File, error) {
+	file, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		err = errors.Wrapf(err, "Cannot open outputFile %v for write", fileName)
```
Comment (tamaseger81) [inline executor/executor.go]: `return nil, errors.Wrapf(err, "Cannot open outputFile %v for write", fileName)`

File: executor/executor.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	}
+	return file, err
+}
+
+// ExecCommand accepts a string as a command (it will be parsed to basecommand and args)
+// and an io.Writer where the result of the execution will be writen (both stdout and stderr).
+// The function may return an error raised during the execution
+func ExecCommand(command string, w io.Writer) (err error) {
```
Comment (tamaseger81) [inline executor/executor.go]: `func ExecCommand(command Command, commandArgs CommandArguments, w io.Writer) (err error)`

File: executor/executor.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+func ParseCommand(command string) (string, []string) {
+	commandSlice := strings.Split(command, " ")
+	name := commandSlice[0]
+	args := commandSlice[1:]
+	return name, args
+}
+
+// GetWriteFile accepts a filename and tries to open it for Writing.
+// It returns a pointer to the file, and an optional error (if it fails to get the required file).
+func GetWriteFile(fileName string) (*os.File, error) {
```
Comment (tamaseger81) [inline executor/executor.go]: `NewFileWriter(path string)`

File: executor/executor.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	writer := bufio.NewWriter(w)
+	defer func() {
+		err = writer.Flush()
+		if err != nil {
+			err = errors.Wrap(err, "Cannot flush buffer")
+			log.Fatal(err)
+		}
+	}()
+
+	baseCommand, args := ParseCommand(command)
```
Comment (tamaseger81) [inline executor/executor.go]: SRP, this function should not be repressible for parsing the command

File: executor/executor_test.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+type CommandExecutorTestSuite struct {
+	suite.Suite
+	*require.Assertions
+}
+
+func (s *CommandExecutorTestSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+}
+
+func (s *CommandExecutorTestSuite) TestParseCommand_empty_input() {
```
Comment (tamaseger81) [inline executor/executor_test.go]: `TestParseCommand_Empty_Input`

File: main.go [hunk 7]
```diff
@@ -18,25 +16,30 @@ var (
...
 		err = outputFile.Close()
 		if err != nil {
-			log.Fatal(err)
+			log.Fatalf("Cannot close outputFile %v\nError :\n%v\n", outputFileName, err)
 		}
 	}()
 
-	err = execCommand(command, outputFile)
+	err = executor.ExecCommand(command, outputFile)
 	if err != nil {
```
Comment (tamaseger81) [inline main.go]:
```
if err := executor.ExecCommand(command, outputFile); err != nil {
```

**Review by tamaseger81 — CHANGES_REQUESTED** [review]

**Comment by mikloslorinczi** [comment]
Implemented all suggested change
Project builds successfully on Bitrise

**Pull request merged**